api.go

```diff
@@ -375,11 +375,16 @@ type Stats struct {
 
 	// Number of times regexp was called on files that we evaluated.
 	RegexpsConsidered int
+
+	// FlushReason explains why results were flushed.
+	FlushReason string
```

Comment:
Only downside this doesn't really aggregate, but I suppose neither does wait time/etc.

Only thing I'd consider (but not block on) is making this API use an enum instead of a string. Or at least documenting the values for FlushReason.

Reply:
> Only downside this doesn't really aggregate, but I suppose neither does wait time/etc.

yup, I agree. The way it is implemented now, the last reason "wins" IE we have to make sure we log the reason for each zeokt before it gets overridden by the aggregator in frontend.

> Only thing I'd consider (but not block on) is making this API use an enum instead of a string. Or at least documenting the values for FlushReason.

Done.